src/map/mob.hpp

```diff
@@ -178,18 +178,36 @@ enum e_aegis_monsterclass : int8 {
 };
 
 struct s_mob_skill {
-	enum MobSkillState state;
+	e_MobSkillState state;
 	uint16 skill_id,skill_lv;
 	short permillage;
 	int casttime,delay;
-	short cancel;
+	bool cancel;
 	short cond1,cond2;
 	short target;
 	int val[5];
```

Comment:
```suggestion
	std::vector<int32> val;
```
Would be great to vectorize this so that we can just remove the limit. If I'm thinking too far ahead that's fine and it can stay as a static array (but maybe increase it to 10 or something).